### ISSUE TITLE
Description of a component is (still) not picked up right by [addon-info]

### ISSUE THREAD

**Issue by marcel-k** [storybookjs/storybook#6264]
**Describe the bug**
Description of a component is (still) not picked up right by addon-info. 

```javascript
var validMatches = [kind, name];
if (validMatches.includes(STORYBOOK_REACT_CLASSES[key].name)) {
  var componentDescription = STORYBOOK_REACT_CLASSES[key].docgenInfo.description;
  retDiv = _react.default.createElement("div", null, _this3.marksy(componentDescription).tree);
}
 ```

The above code in Story.js function `_getComponentDescription` only work when:
- The storiesOf 'name' parameter (`kind`) is exact match
- The story.add 'storyName' parameter is an exact match (when would you want to do this?)

In my situation, I want to create hierarchy by using slashes in the storiesOf 'name' and I would use the storyName to describe the test/state of the component in the story, like so:

```javascript
const stories = storiesOf('atoms/button/ButtonPrimary', module);
stories.add('default', () => <ButtonPrimary /> );
stories.add('some prop', () => <ButtonPrimary  someProp={true} />);
```

suggested solution would be to get the `hierarchySeparator` and split the storyKind with it and only use the last entree in the comparison, if that is possible and doesn't break anything:

```javascript
var realKind = kind.split(hierarchySeparator || '/');
var validMatches = [realKind[realKind.length - 1], name];
if (validMatches.includes(STORYBOOK_REACT_CLASSES[key].name)) {
  var componentDescription = STORYBOOK_REACT_CLASSES[key].docgenInfo.description;
  retDiv = _react.default.createElement("div", null, _this3.marksy(componentDescription).tree);
}
```

"react": "^16.8.4",
"@storybook/react": "^5.0.5",
"@storybook/addon-info": "^5.0.5"

**Comment by marcel-k** [comment]
Even with this fix, it still won't work with styled components though, neither does the proptable. related to #6345 ?

**Comment by marcel-k** [comment]
Will this be fixed in 5.1.0?

**Comment by shilman** [comment]
We're generally focusing on addon-docs, a replacement for addon-info that fixes this issue and many others. 

https://medium.com/storybookjs/storybook-docs-sneak-peak-5be78445094a

It's currently usable in technical preview in the `#docs-mode` channel on our Discord https://discord.gg/UUt2PJb and will be documented as part of the 5.1 release (but officially released in 5.2)
